### PR TITLE
feat(cache): Add cache validator

### PR DESCRIPTION
**Pull request by kentcdodds**
This one should have two reviewers I think. I've added a `utils` directory for utility functions to use throughout the codebase. The first of these is `friendly-type-of` which gives us more helpful output than `typeof []` or `typeof new RegExp()` :-)

### CODE REVIEW

**Comment by sarbbottam** [comment]
:+1: LGTM

File: src/validators/cache/index.js [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
 export default {
   key: 'cache',
-  validate: noop,
+  validate: validateCache,
+}
+
+function validateCache(value) {
+  const type = friendlyTypeOf(value)
+  if (type !== 'boolean' && type !== 'object') {
+    return `Unexpected type ${type}. Must be a boolean or an object.`
```
Comment (jonathanewerner) [inline src/validators/cache/index.js]: So the style guide for messages I formulated says no period at the end. Reason: when you have multiple errors joined by " - ", periods look awkward in front of the dash. I'm torn though, as I like it better with periods, apart from that corner case. What do you think?

Comment (kentcdodds) [inline src/validators/cache/index.js]: Oh yeah, sorry! Will fix :-)

Comment (jonathanewerner) [inline src/validators/cache/index.js]: Apart from that: 👍🏻. Will hit the green button as soon as this is done so you can use the shiny util fn in the other PR. ;)

**Pull request merged**